polkadot/runtime/common/src/claims.rs

```diff
@@ -611,46 +619,45 @@ impl<T: Config> PrevalidateAttests<T>
 where
 	<T as frame_system::Config>::RuntimeCall: IsSubType<Call<T>>,
 {
-	/// Create new `SignedExtension` to check runtime version.
+	/// Create new `TransactionExtension` to check runtime version.
 	pub fn new() -> Self {
 		Self(core::marker::PhantomData)
 	}
 }
 
-impl<T: Config> SignedExtension for PrevalidateAttests<T>
+impl<T: Config> TransactionExtensionBase for PrevalidateAttests<T>
```

Comment:
one annoying part about `feeless_if` is we do the check 2 times.
(and IMHO also that signer can't opt-out of fees in case of failing transaction).

Ideally the `feeless_if` check should add information into the origin. And then the call can avoid the second check if it is already feeless.

But currently origins don't have any mechanism to add information.

* We could have a transaction extension which change to a new origin `FeelessSignedOrigin(AccountId)` but then this origin would bypass the `Nonce` check.
  So maybe we can change to a new origin `FeelessSignedOrigin(AccountId, Nonce)` and make the `Nonce` of an origin retrievable through a trait.
* Or we can make some change to the origin to be able to have additional information added in the pipeline. Like an origin wil be:
  * a caller
  * a filter
  * `Vec<Vec<u8>>` a set of additional information encoded.

All in all I think `feeless_if` story should be improved. So maybe better to keep this as a transaction extension, and upgrade it when `feeless_if` get reworked.